modules/platform/nuxeo-automation/nuxeo-automation-features/src/test/java/org/nuxeo/ecm/automation/core/operations/coldstorage/AbstractTestColdStorageOperation.java

```diff
@@ -66,18 +68,52 @@ protected void moveContentToColdStorage(CoreSession session, DocumentModel docum
         try (OperationContext context = new OperationContext(session)) {
             context.setInput(documentModel);
             DocumentModel updatedDocModel = (DocumentModel) automationService.run(context, MoveToColdStorage.ID);
+            checkMoveContent(List.of(documentModel), List.of(updatedDocModel));
+        }
+    }
+
+    protected void moveContentToColdStorage(CoreSession session, List<DocumentModel> documents)
+            throws OperationException, IOException {
+        List<String> documentIds = documents.stream().map(DocumentModel::getId).collect(Collectors.toList());
+        try (OperationContext context = new OperationContext(session)) {
+            context.setInput(documents);
+            checkMoveContent(documents, (DocumentModelList) automationService.run(context, MoveToColdStorage.ID));
+        }
+    }
+
+    protected void checkMoveContent(List<DocumentModel> expectedDocs, List<DocumentModel> foundedDocs)
+            throws IOException {
+        assertEquals(expectedDocs.size(), foundedDocs.size());
+        List<String> expectedDocIds = expectedDocs.stream().map(DocumentModel::getId).collect(Collectors.toList());
+        for (DocumentModel updatedDocModel : foundedDocs) {
             Blob fileContent = (Blob) updatedDocModel.getPropertyValue(ColdStorageHelper.FILE_CONTENT_PROPERTY);
             Blob coldStorageContent = (Blob) updatedDocModel.getPropertyValue(
                     ColdStorageHelper.COLD_STORAGE_CONTENT_PROPERTY);
-            assertEquals(documentModel.getRef(), updatedDocModel.getRef());
-            assertTrue(updatedDocModel.hasFacet(FacetNames.COLD_STORAGE));
-            assertEquals(DummyThumbnailFactory.DUMMY_THUMBNAIL_CONTENT, fileContent.getString());
-            assertEquals(FILE_CONTENT, coldStorageContent.getString());
+
+            // check document
+            assertTrue(String.format("Doc: %s must one of %s", updatedDocModel, expectedDocs),
+                    expectedDocIds.contains(updatedDocModel.getId()));
+            assertTrue(String.format("Doc: %s must have the %s Facet", updatedDocModel, FacetNames.COLD_STORAGE),
+                    updatedDocModel.hasFacet(FacetNames.COLD_STORAGE));
+
+            // check blobs
+            assertEquals(String.format("Doc: %s must have %s in 'file:content'", //
+                    updatedDocModel, DummyThumbnailFactory.DUMMY_THUMBNAIL_CONTENT), //
+                    DummyThumbnailFactory.DUMMY_THUMBNAIL_CONTENT, fileContent.getString());
+            assertEquals(String.format("Doc: %s must have %s in 'coldstorage:coldContent'", //
+                    updatedDocModel, FILE_CONTENT), //
+                    FILE_CONTENT, coldStorageContent.getString());
```

Comment:
The `assertEquals` messages are not really needed, and it makes the code/test hardly understandable IMO.

Reply:
yes, ;) it is mainly what I did in the past, but since I had a discussion wiht Florent on the assertion in theses cases (loop) he advised me to add a message to make easier the comprehension of things when the test failed. but no pbs if we all agree i can remove them :)

Reply:
I took into account your suggestion :), I let the conversation open to let more feedbacks

Reply:
I mark the conversation as resolved, I made the modification to suit your suggestions. I thinks these kind of interrogation; static import, message in assertion will be more global and I think will be part of the guideline that you are going to do :)